src/playlist.js

```diff
@@ -105,11 +105,12 @@
       if (!playlist.endList) {
         return window.Infinity;
       }
+      endSequence = playlist.segments.length;
```

Comment:
I have tried to consistently use the suffix "sequence" only for variables that store numbers in the media sequence number line. Looks like you're using it for an available segment index here.